src/python/pants/python/python_setup.py

```diff
@@ -32,12 +32,26 @@ def register_options(cls, register):
             type=list,
             default=["CPython>=3.6"],
             metavar="<requirement>",
-            help="Constrain the selected Python interpreter.  Specify with requirement syntax, "
+            help="Constrain the selected Python interpreter. Specify with requirement syntax, "
             "e.g. 'CPython>=2.7,<3' (A CPython interpreter with version >=2.7 AND version <3)"
             "or 'PyPy' (A pypy interpreter of any version). Multiple constraint strings will "
             "be ORed together. These constraints are applied in addition to any "
             "compatibilities required by the relevant targets.",
         )
+        register(
+            "--requirement-constraints",
+            advanced=True,
+            fingerprint=True,
+            # TODO(#9148): uncomment out the below line once this is fixed. Right now, using
+            # `file_option` causes the unit test to fail.
+            # type=file_option,
```

Comment:
This is my next priority to fix. See https://github.com/pantsbuild/pants/issues/9148. The fix is fairly simple, it's more a question of how to do it cleanly.

Reply:
See https://github.com/pantsbuild/pants/pull/9190#pullrequestreview-365492806.